hibernate-core/src/main/java/org/hibernate/loader/BatchLoadSizingStrategy.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.loader;
+
+/**
+ * Strategy (pluggable) for determining an optimal size for batch loads.
+ *
+ * @author Steve Ebersole
+ */
+public interface BatchLoadSizingStrategy {
+	int determineOptimalBatchLoadSize(int numberOfKeyColumns, int numberOfKeys);
```

Comment:
> Yes, I do it where I think it helps reviewers to better understand my draft.

When the same points are already discussed in the Jira?  Ok, bonus points to you I guess...

> Introducing a parameter object may be a good idea with respect to evolution of the API (hint: this is a proposal for an alternative ;)

If we were to go that route, I'd actually just move these from `IdentifierLoadAccess` into a new contract (`MultIdentifierLoadAccess`?)  Ultimately it is the same principle, the difference being that the user isn't having to instantiate that extra object directly.